app/guided.py

```diff
@@ -0,0 +1,424 @@
+        else:
+            planner_goals = goals[:]
+
+        await self._show_attack_graph(attack_graph) # TODO: Remove before merging, just for debug purposes
+
+        absolute_distance_table = await self._build_distance_table(attack_graph, goals=planner_goals)
+        effective_distance_table = copy.deepcopy(absolute_distance_table)
+
+        goal_links = await self._get_goal_links(absolute_distance_table, agent=agent)
+        while len(planner_goals) > 0 and len(goal_links) > 0:
+            tasked_links = await self._task_agents(goal_links, effective_distance_table, abilities, agent=agent)
+
+            await self.operation.wait_for_links_completion(tasked_links)
+            
+            current_goal_count = len(planner_goals)
+            planner_goals = await self._update_goals(current_goals=planner_goals)
+            if len(planner_goals) < current_goal_count:
+                absolute_distance_table, effective_distance_table = await self._handle_goal_achieved(attack_graph, goals=planner_goals)
+            else:
+                effective_distance_table = await self._handle_no_goal_achieved(absolute_distance_table,
+                                                                                effective_distance_table,
+                                                                                goal_links)
+            goal_links = await self._get_goal_links(absolute_distance_table, agent=agent)
+
+        if not len(goal_links) or not len(planner_goals):
+            self.planning_svc.log.debug('No more links available or all goals satisfied, planner complete.')
+            self.next_bucket = None
+
+    """ PRIVATE """
+
+    async def _build_attack_graph(self, abilities, agent):
+        """
+        Produces a directed graph that includes nodes for each ability, and nodes of each input and output fact
+        of those abilities.
+        """
+        attack_graph = nx.DiGraph()
+        for ability_id in abilities:
+            for ability in await self.data_svc.locate('abilities', dict(ability_id=ability_id)):
+                if agent:
+                    attack_graph = await self._output_fact_edges(ability, attack_graph, agent)
+                    attack_graph = await self._input_fact_edges(ability, attack_graph, agent)
+                else:
+                    for operation_agent in self.operation.agents:
+                        attack_graph = await self._output_fact_edges(ability, attack_graph, operation_agent)
+                        attack_graph = await self._input_fact_edges(ability, attack_graph, operation_agent)
+        return attack_graph
+
+    async def _output_fact_edges(self, ability, graph, agent):
+        """
+        Adds output facts to a directed graph based on the parserconfigs included in an ability.
+        """
+        executor = await agent.get_preferred_executor(ability)
+        for parser in executor.parsers:
```

Comment:
If an ability with no executor for this agent/platform is encountered (e.g., Net use on Darwin) this will raise an attribute error.

'NoneType' object has no attribute 'parsers'

I believe the desired behavior would be to skip the ability if there is no executor.